src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SuppressCommentFix.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Acuminator.Utilities.Common;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Acuminator.Analyzers.StaticAnalysis
+{
+	[Shared]
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class SuppressCommentFix : CodeFixProvider
+	{
+		private const string _comment = @"// Acuminator disable once {0} {1} [Justification]";
+		private const string _diagnosticName = @"Suppress diagnostic {0}";
+
+		public static readonly ImmutableDictionary<string, string> _fixableDiagnosticIds;
+
+		static SuppressCommentFix()
+		{
+			Type diagnosticsType = typeof(Descriptors);
+			var propertiesInfo = diagnosticsType.GetRuntimeProperties();
+			
+			_fixableDiagnosticIds = propertiesInfo
+				.Where(x => x.PropertyType == typeof(DiagnosticDescriptor))
+				.Select(x =>
+				{
+					var descriptor = (DiagnosticDescriptor) x.GetValue(x, null);
+					return new
+					{
+						descriptor.Id,
+						Name = descriptor
+							.CustomTags.FirstOrDefault()
+							.CheckIfNullOrWhiteSpace(nameof(descriptor.CustomTags))
+					};
+				}).Distinct()
+				.ToImmutableDictionary(x => x.Id, x => x.Name);
+		}
+
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			_fixableDiagnosticIds.Keys.ToImmutableArray();
+
+		public override Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			return Task.Run(() =>
+			{
+				Parallel.ForEach(context.Diagnostics, (diagnostic) => {
```

Comment:
Parallel.ForEach also has an overload that accepts options in which you can specify a `cancellationToken`. Look for analyzer aggregators for graph, events or DAC